storage/innobase/row/row0ins.cc

```diff
@@ -2144,13 +2144,11 @@ row_ins_scan_sec_index_for_duplicate(
 	n_unique first fields is NULL, a unique key violation cannot occur,
 	since we define NULL != NULL in this case */
 
-	if (!index->nulls_equal) {
-		for (ulint i = 0; i < n_unique; i++) {
-			if (UNIV_SQL_NULL == dfield_get_len(
-					dtuple_get_nth_field(entry, i))) {
+	for (ulint i = 0; i < n_unique; i++) {
+		if (UNIV_SQL_NULL == dfield_get_len(
+				dtuple_get_nth_field(entry, i))) {
```

Comment:
This could be written as `dfield_is_null(dtuple_get_nth_field(entry, i))`. In `btr_pcur_t::restore_position()` but not here we only enter the loop if `index->n_nullable` is nonzero. Maybe, invoke a common function for both?

Reply:
I added `index->n_nullable` check in the condition. I did not include this condition in `dfield_is_null()` because in this case we would need to pass index into the function, and the only reason for such passing is to quit the function if some member of index is 0 at the very beginning of the function, i.e. if the condition is false, we should not invoke the function at all.